### ISSUE TITLE
After GUI merge: Add callbacks system

### ISSUE THREAD

**Issue by Dunbaratu** [KSP-KOS/KOS#1949]
(This is one of several issues that should be addressed after #1878 (GUI toolkit) gets merged into develop but before that code appears in a public realease.)

Right now the GUI toolkit has to have kerobscript code that busy-polls for the status of things like buttons and the like.   I had an experimental branch that allows C# code to schedule the execution of kerbscript callbacks but we didn't have a good use case for it so I never made a PR for it.

Now we do have a very good use case for it - now we can have the kerboscript program give callback hooks for the GUI system to call when widgets are used.  (i..e things like an `OnClick` for a button, or an `OnChange` for a selection list, etc.)

So, after the GUI system is merged into develop, we need to get that callbacks system merged in too, and then modify the GUI system to use the callbacks.  This issue is for reminding us to do that before release.
